external/anomaly/ote_anomalib/callbacks/progress.py

```diff
@@ -0,0 +1,123 @@
+        super().on_predict_start(trainer, pl_module)
+        self._reset_progress()
+
+    def on_test_start(self, trainer, pl_module):
+        super().on_test_start(trainer, pl_module)
+        self._reset_progress()
+
+    def on_train_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
+        """
+        Adds training completion percentage to the progress bar
+        """
+        super().on_train_batch_end(trainer, pl_module, outputs, batch, batch_idx, dataloader_idx)
+        self.current_epoch = trainer.current_epoch
+        self._update_progress(stage="train")
+
+    def on_validation_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
+        """
+        Adds validation completion percentage to the progress bar
+        """
+        super().on_validation_batch_end(trainer, pl_module, outputs, batch, batch_idx, dataloader_idx)
+        self._update_progress(stage="val")
+
+    def on_predict_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
+        """
+        Adds prediction completion percentage to the progress bar
+        """
+        super().on_predict_batch_end(trainer, pl_module, outputs, batch, batch_idx, dataloader_idx)
+        self._update_progress(stage="predict")
+
+    def on_test_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
+        """
+        Adds testing completion percentage to the progress bar
+        """
+        super().on_test_batch_end(trainer, pl_module, outputs, batch, batch_idx, dataloader_idx)
+        self._update_progress(stage="test")
+
+    def _reset_progress(self):
+        self._progress = 0.0
+
+    def _get_progress(self, stage: str = "train") -> float:
+        """
+        Get progress for train and test stages.
+
+        Args:
+            stage (str, optional): Train or Test stages. Defaults to "train".
+        """
+
+        if stage == "train":
+            # Progress is calculated on the upper bound (max epoch).
+            # Early stopping might stop the training before the progress reaches 100%
+            self._progress = (
+                (self.train_batch_idx + self.current_epoch * self.total_train_batches)
+                / (self.total_train_batches * self.max_epochs)
+            ) * 100
+
+        elif stage == "val":
+            self._progress = (self.val_batch_idx / (self.total_val_batches + 1e-10)) * 100
+
+        elif stage == "predict":
+            self._progress = (self.predict_batch_idx / (self.total_predict_batches + 1e-10)) * 100
+
+        elif stage == "test":
+            self._progress = (self.test_batch_idx / (self.total_test_batches + 1e-10)) * 100
+        else:
+            raise ValueError(f"Unknown stage {stage}. Available: train, val, predict and test")
+
+        return self._progress
+
+    def _update_progress(self, stage: str):
+        progress = self._get_progress(stage)
+        self.update_progress_callback(progress)
```

Comment:
Until we fix the SDK please make sure that you pass progress value as an integer.